### PR TITLE
ICU-21668 Fix nickel rounding with away-from-zero rounding mode

### PR DESCRIPTION
**Pull request by sffc**
CC @anba

##### Checklist

- [x] Required: Issue filed: https://unicode-org.atlassian.net/browse/ICU-21668
- [x] Required: The PR title must be prefixed with a JIRA Issue number. 
- [x] Required: The PR description must include the link to the Jira Issue, for example by completing the URL in the first checklist item
- [x] Required: Each commit message must be prefixed with a JIRA Issue number. 
- [x] Issue accepted (done by Technical Committee after discussion)
- [x] Tests included, if applicable
- [ ] API docs and/or User Guide docs changed or added, if applicable

### CODE REVIEW

File: icu4c/source/test/intltest/numbertest_api.cpp [hunk 1]
```diff
@@ -3180,6 +3180,78 @@ void NumberFormatterApiTest::roundingOther() {
             u"0.000",
             u"0.000");
 
+    assertFormatDescending(
```
Comment (FrankYFTang) [inline icu4c/source/test/intltest/numbertest_api.cpp:3183]: could you also add a test case of what anba put into https://unicode-org.atlassian.net/browse/ICU-21668 here so we can show the code fix truly fix that exact case?

```
auto increment = NumberFormatter::with()
      .precision(Precision::increment(5000).withMinFraction(0))
      .roundingMode(UNUM_ROUND_UP)
      .locale(Locale::getEnglish())
      .formatDouble(5.625, status)
      .toString(status);

```

**Comment by sffc** [comment]
This PR depends on #1813

File: icu4c/source/test/intltest/numbertest_api.cpp (hunk 1 above)
Comment (FrankYFTang) [reply icu4c/source/test/intltest/numbertest_api.cpp:3183]: waiting for that particular exact test case to be added.

Comment (FrankYFTang) [reply icu4c/source/test/intltest/numbertest_api.cpp:3183]: waiting that exact test case to be added

Comment (sffc) [reply icu4c/source/test/intltest/numbertest_api.cpp:3184]: added.

**Review by FrankYFTang — APPROVED** [review]
thanks LGTM

**Review by FrankYFTang — APPROVED** [review]
